examples/audio/soundtrack.rs

```diff
@@ -0,0 +1,154 @@
+    commands.insert_resource(SoundtrackPlayer::new(track_list));
+}
+
+
+// Every time the GameState resource changes, this system is run to trigger the song change.
+fn change_track(
+    mut commands: Commands,
+    sountrack_player: Res<SoundtrackPlayer>,
+    soundtrack: Query<Entity, With<AudioSink>>,
+    game_state: Res<GameState>,
+) {
+    // Fade out all currently running tracks
+    for track in soundtrack.iter() {
+        commands.entity(track).insert(FadeOut);
+    }
+
+    // Spawn a new `AudioBundle` with the appropriate soundtrack based on 
+    // the game state.
+    //
+    // Volume is set to start at zero and is then increased by the fade_in system.
+    match game_state.as_ref() {
+        GameState::Peaceful => {
+            commands.spawn((
+                    AudioBundle {
+                        source: sountrack_player.track_list.get(0).unwrap().clone(),
+                        settings: PlaybackSettings {
+                            mode: bevy::audio::PlaybackMode::Loop,
+                            volume: bevy::audio::Volume::ZERO,
+                            ..default()
+                        }
+                    }, 
+                    FadeIn,
+            ));
+        }
+        GameState::Battle => {
+            commands.spawn((
+                    AudioBundle {
+                        source: sountrack_player.track_list.get(1).unwrap().clone(),
+                        settings: PlaybackSettings {
+                            mode: bevy::audio::PlaybackMode::Loop,
+                            volume: bevy::audio::Volume::ZERO,
+                            ..default()
+                        }
+                    }, 
+                    FadeIn,
+            ));
+        }
+    }
+}
+
+// Fade effect duration
+const FADE_TIME: f32 = 2.0;
+
+// Fades in the audio of entities that has the FadeIn component. Removes the FadeIn component once
+// full volume is reached.
+fn fade_in(
+    mut commands: Commands,
+    mut audio_sink: Query<(&mut AudioSink, Entity), With<FadeIn>>,
+    time: Res<Time>,
+) {
+    for (audio, entity) in audio_sink.iter_mut() {
+        audio.set_volume(audio.volume() + time.delta_seconds() / FADE_TIME);
+        if audio.volume() >= 1.0 {
+            audio.set_volume(1.0);
+            commands.entity(entity).remove::<FadeIn>();
+        }
+    }
+}
+
+// Fades out the audio of entities that has the FadeOut component. Despawns the entities once audio
+// volume reaches zero.
+fn fade_out(
+    mut commands: Commands,
+    mut audio_sink: Query<(&mut AudioSink, Entity), With<FadeOut>>,
+    time: Res<Time>,
+) {
+    for (audio, entity) in audio_sink.iter_mut() {
+        audio.set_volume(audio.volume() - time.delta_seconds() / FADE_TIME);
+        if audio.volume() <= 0.0 {
+            commands.entity(entity).despawn_recursive();
+        }
+    }
+}
+
+// Every time the timer ends, switches between the "Peaceful" and "Battle" state.
```

Comment:
I think it would be nicer to instead react to a user input for the transition, just to really highlight how this is a dynamic system, not just a single audio file with built in fade transitions.

Might also be nice to change the clear colour of the window between a "battle" colour and a "peaceful" colour.

Reply:
I did think about using keyboard input. It is certainly something that can be modified.

I do like the window changing color! But do you think it would stay in the spirit of keeping the examples focused on a single mechanic/system?

Reply:
Yeah, IMO we should keep it minimal.